### ISSUE TITLE
Some tests in test_cert.py fail with requests 2.32

### ISSUE THREAD

**Issue by yan12125** [hvac/hvac#1177]
As title, some tests fail with:
```
>               context.load_cert_chain(certfile, keyfile)
E               urllib3.exceptions.ProtocolError: ('Connection aborted.', FileNotFoundError(2, 'No such file or directory'))
```
Complete logs can be found at https://github.com/yan12125/hvac/actions/runs/9612728771, where I [updated `poetry.lock`](https://github.com/yan12125/hvac/commit/743dddf4bb74b512aa28264495f6c5dda45af149) to test newer requests.

Requests 2.32 comes with [some changes in SSL](https://github.com/psf/requests/blob/main/HISTORY.md#2320-2024-05-20). Maybe the way those tests specify certificates or key files are no longer compatible.

**Comment by Tylerlhess** [comment]
Yeah I can't seem to get this failure with any version of requests 2.32.0-2.32.4

**Comment by yan12125** [comment]
Thank you for revisiting this issue, while I may no longer work on it. Hvac integration tests are broken from time to time when each new vault version is out (ex: https://github.com/hvac/hvac/issues/786, https://github.com/hvac/hvac/issues/905, https://github.com/hvac/hvac/issues/1152), and brought me maintenance burden on the [python-hvac](https://archlinux.org/packages/extra/any/python-hvac/) Arch Linux package. To mitigate packaging efforts, I [disabled integration tests altogether](https://gitlab.archlinux.org/archlinux/packaging/packages/python-hvac/-/commit/013f1ce4cf170cd0b2be30fa71cfac7e457016f5), and thus `tests/integration_tests/api/auth_methods/test_cert.py` is no longer tested. Feel free to close the issue if you wish.

> As this seems to have failed in the CI and the certificate files are dynamically generated during the testing could this error be arising from a permissions issue on the CI platform where the files can't be created?

I may not be a CI platform issue. As a record, when I reported this issue last year, I got the issue locally first and then reproduced it on CI.
